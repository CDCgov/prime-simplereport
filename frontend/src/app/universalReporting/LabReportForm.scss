.lab-combo-box {
  max-width: none !important;
}

.usa-form-group.report-form-controls {
  margin-top: 18px;
}
