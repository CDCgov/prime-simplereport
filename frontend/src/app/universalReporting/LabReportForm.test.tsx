import { fireEvent, render, screen, within } from "@testing-library/react";
import configureStore, { MockStoreEnhanced } from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";
import userEvent from "@testing-library/user-event";

import {
  GetAllLabsDocument,
  GetSpecimensByLoincDocument,
} from "../../generated/graphql";

import LabReportForm from "./LabReportForm";

window.scrollTo = jest.fn();

describe("LabReportForm", () => {
  let store: MockStoreEnhanced<unknown, {}>;
  const mockStore = configureStore([]);
  const renderWithUser = (mocks: any[]) => ({
    user: userEvent.setup(),
    ...render(
      <MemoryRouter>
        <MockedProvider mocks={mocks}>
          <Provider store={store}>
            <LabReportForm />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    ),
  });

  beforeEach(() => {
    store = mockStore({
      organization: {
        name: "Organization Name",
      },
      facilities: [
        {
          id: "a1",
          name: "Fake Facility",
        },
      ],
      user: {
        permissions: [],
      },
    });
  });

  test("renders stepper with steps", () => {
    renderWithUser([]);
    const stepIndicator = screen.getByTestId("step-indicator");
    const steps = within(stepIndicator).getAllByRole("listitem");
    expect(steps).toHaveLength(5);
    expect(steps[0]).toHaveTextContent("Facility information");
    expect(steps[1]).toHaveTextContent("Provider information");
    expect(steps[2]).toHaveTextContent("Patient information");
    expect(steps[3]).toHaveTextContent("Lab results");
    expect(steps[4]).toHaveTextContent("Review and submit");

    const nextButton = screen.getByRole("button", {
      name: /Next/i,
    });

    expect(nextButton).toHaveTextContent("Next: Provider information");
  });

  test("steps forward and back", () => {
    renderWithUser([]);

    const stepIndicator = screen.getByTestId("step-indicator");

    const currentStep = within(stepIndicator).getByRole("heading");
    expect(currentStep).toHaveTextContent("Facility information");

    const nextButton = screen.getByRole("button", {
      name: /Next/i,
    });
    fireEvent.click(nextButton);

    expect(currentStep).toHaveTextContent("Provider information");

    const backButton = screen.getByRole("button", {
      name: /Back to facility information/i,
    });
    fireEvent.click(backButton);

    expect(currentStep).toHaveTextContent("Facility information");
  });

  test("allows clicking to change step", async () => {
    renderWithUser([]);

    const stepIndicator = screen.getByTestId("step-indicator");
    const currentStep = within(stepIndicator).getByRole("heading");

    const providerStep = within(stepIndicator).getByText(
      "Provider information"
    );
    fireEvent.click(providerStep);
    expect(currentStep).toHaveTextContent("Provider information");

    const patientStep = within(stepIndicator).getByText("Patient information");
    fireEvent.click(patientStep);
    expect(currentStep).toHaveTextContent("Patient information");
    await confirmPatientFormSectionFieldNames();

    const facilityStep = within(stepIndicator).getByText(
      "Facility information"
    );
    fireEvent.click(facilityStep);
    expect(currentStep).toHaveTextContent("Facility information");

    fireEvent.click(patientStep);
    await confirmPatientFormFieldNamesSaved();
  });

  describe("LabResultsFormSection", () => {
    it("renders lab results page successfully", async () => {
      const { user } = renderWithUser(labSectionMocks);

      const stepIndicator = screen.getByTestId("step-indicator");
      const resultsStep = within(stepIndicator).getByText("Lab results");
      await user.click(resultsStep);

      expect(
        await screen.findByText("Choose a specimen type to enter test results")
      );
      expect(
        await screen.findByText(
          "Please select test order and specimen type above to see test result information"
        )
      );
    });

    it("renders lab results state for no test orders found", async () => {
      const noLabsMock = [
        {
          request: {
            query: GetAllLabsDocument,
          },
          result: {
            data: {
              labs: [],
            },
          },
        },
      ];

      const { user } = renderWithUser(noLabsMock);

      const stepIndicator = screen.getByTestId("step-indicator");
      const resultsStep = within(stepIndicator).getByText("Lab results");
      await user.click(resultsStep);

      expect(
        await screen.findByText(
          "No test orders found. Please contact support for assistance."
        )
      );
    });

    it("renders test result fields when test order and specimen are selected", async () => {
      const { user, container } = renderWithUser(labSectionMocks);

      const stepIndicator = screen.getByTestId("step-indicator");
      const resultsStep = within(stepIndicator).getByText("Lab results");
      await user.click(resultsStep);

      expect(
        await screen.findByText("Choose a specimen type to enter test results")
      );

      await user.click(screen.getByRole("combobox", { name: "Test order" }));
      await user.click(
        screen.getByRole("option", {
          name: "Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay",
        })
      );

      await user.selectOptions(
        screen.getByRole("combobox", { name: "Specimen type" }),
        "Serum specimen from blood product - 122591000"
      );

      expect(
        await screen.findByText(
          "The lab order and specimen you selected has one corresponding test result associated with it. Please fill out the test result details below."
        )
      );
      expect(
        screen.getByRole("group", { name: "Test result value" })
      ).toBeVisible();

      expect(container).toMatchSnapshot();
    });
  });
});
const confirmPatientFormSectionFieldNames = async () => {
  const patientFormSection = screen.getByTestId("patientFormSection");
  const textBoxInputs = within(patientFormSection).getAllByRole("textbox");
  const dropdownInputs = within(patientFormSection).getAllByRole("combobox");
  expect(textBoxInputs).toHaveLength(11);
  expect(dropdownInputs).toHaveLength(3);

  const firstNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-first-name"
  )[0];
  const middleNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-middle-name"
  )[0];
  const lastNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-last-name"
  )[0];
  const patientIdInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-id"
  )[0];
  const streetOneInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-street-one"
  )[0];
  const streetTwoInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-street-two"
  )[0];
  const cityInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-city"
  )[0];
  const countyInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-county"
  )[0];
  const zipCodeInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-zip-code"
  )[0];
  const phoneInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-phone"
  )[0];
  const emailInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-email"
  )[0];
  // const tribalAffiliationDropdown = dropdownInputs.filter(
  //   (input) => input.getAttribute("id") === "tribal-affiliation"
  // )[0];
  // const countryDropdown = dropdownInputs.filter(
  //   (input) => input.getAttribute("name") === "patient-country"
  // )[0];
  // const stateDropdown = dropdownInputs.filter(
  //   (input) => input.getAttribute("name") === "patient-state"
  // )[0];

  await userEvent.click(firstNameInput);
  await userEvent.keyboard("Firstly");
  await userEvent.click(middleNameInput);
  await userEvent.keyboard("Middleton");
  await userEvent.click(lastNameInput);
  await userEvent.keyboard("Lastly");
  await userEvent.click(patientIdInput);
  await userEvent.keyboard("My Custom Id");
  await userEvent.click(streetOneInput);
  await userEvent.keyboard("My Street 1");
  await userEvent.click(streetTwoInput);
  await userEvent.keyboard("My Street 2");
  await userEvent.click(cityInput);
  await userEvent.keyboard("Baltimore");
  await userEvent.click(countyInput);
  await userEvent.keyboard("Fulton");
  await userEvent.click(zipCodeInput);
  await userEvent.keyboard("30319");
  await userEvent.click(phoneInput);
  await userEvent.keyboard("555-555-5555");
  await userEvent.click(emailInput);
  await userEvent.keyboard("fakeemail@example.com");

  const femaleRadioOption = screen.getAllByLabelText("Female")[0];
  expect(femaleRadioOption).not.toBeChecked();
  await userEvent.click(femaleRadioOption);
  expect(femaleRadioOption).toBeChecked();

  const nativeRadioOption = screen.getAllByLabelText(
    "Native Hawaiian/other Pacific Islander"
  )[0];
  expect(nativeRadioOption).not.toBeChecked();
  await userEvent.click(nativeRadioOption);
  expect(nativeRadioOption).toBeChecked();

  const latinoYesRadioOption = screen.getAllByLabelText("Yes")[0];
  expect(latinoYesRadioOption).not.toBeChecked();
  await userEvent.click(latinoYesRadioOption);
  expect(latinoYesRadioOption).toBeChecked();

  expect(firstNameInput).toHaveValue("Firstly");
  expect(middleNameInput).toHaveValue("Middleton");
  expect(lastNameInput).toHaveValue("Lastly");
  expect(patientIdInput).toHaveValue("My Custom Id");
  expect(streetOneInput).toHaveValue("My Street 1");
  expect(streetTwoInput).toHaveValue("My Street 2");
  expect(cityInput).toHaveValue("Baltimore");
  expect(countyInput).toHaveValue("Fulton");
  expect(zipCodeInput).toHaveValue("30319");
  expect(phoneInput).toHaveValue("555-555-5555");
  expect(emailInput).toHaveValue("fakeemail@example.com");

  expect(patientFormSection).toHaveTextContent("First name");
  expect(patientFormSection).toHaveTextContent("Middle name (optional)");
  expect(patientFormSection).toHaveTextContent("Last name");
  expect(patientFormSection).toHaveTextContent("Date of birth");
  expect(patientFormSection).toHaveTextContent("Patient ID (optional)");
  expect(patientFormSection).toHaveTextContent("Sex");
  expect(patientFormSection).toHaveTextContent("Female");
  expect(patientFormSection).toHaveTextContent("Male");
  expect(patientFormSection).toHaveTextContent("Race");
  expect(patientFormSection).toHaveTextContent(
    "American Indian/Alaskan Native"
  );
  expect(patientFormSection).toHaveTextContent("Asian");
  expect(patientFormSection).toHaveTextContent("Black/African American");
  expect(patientFormSection).toHaveTextContent(
    "Native Hawaiian/other Pacific Islander"
  );
  expect(patientFormSection).toHaveTextContent("White");
  expect(patientFormSection).toHaveTextContent("Other");
  const unknownLabels = screen.getAllByText("Unknown");
  expect(unknownLabels).toHaveLength(2);
  expect(patientFormSection).toHaveTextContent(
    "Is the patient Hispanic or Latino?"
  );
  expect(patientFormSection).toHaveTextContent("Yes");
  expect(patientFormSection).toHaveTextContent("No");
  expect(patientFormSection).toHaveTextContent("Tribal affiliation (optional)");
  expect(patientFormSection).toHaveTextContent("Patient contact");
  expect(patientFormSection).toHaveTextContent("Street address");
  expect(patientFormSection).toHaveTextContent("Apt, suite, etc (optional)");
  expect(patientFormSection).toHaveTextContent("City");
  expect(patientFormSection).toHaveTextContent("County (optional)");
  expect(patientFormSection).toHaveTextContent("Country");
  expect(patientFormSection).toHaveTextContent("United States");
  expect(patientFormSection).toHaveTextContent("State");
  expect(patientFormSection).toHaveTextContent("ZIP code");
  expect(patientFormSection).toHaveTextContent("Phone number (optional)");
  expect(patientFormSection).toHaveTextContent("Email address (optional)");
};

const confirmPatientFormFieldNamesSaved = async () => {
  const patientFormSection = screen.getByTestId("patientFormSection");
  const textBoxInputs = within(patientFormSection).getAllByRole("textbox");
  const dropdownInputs = within(patientFormSection).getAllByRole("combobox");
  expect(textBoxInputs).toHaveLength(11);
  expect(dropdownInputs).toHaveLength(3);

  const firstNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-first-name"
  )[0];
  const middleNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-middle-name"
  )[0];
  const lastNameInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-last-name"
  )[0];
  const patientIdInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-id"
  )[0];
  const streetOneInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-street-one"
  )[0];
  const streetTwoInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-street-two"
  )[0];
  const cityInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-city"
  )[0];
  const countyInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-county"
  )[0];
  const zipCodeInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-zip-code"
  )[0];
  const phoneInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-phone"
  )[0];
  const emailInput = textBoxInputs.filter(
    (input) => input.getAttribute("name") === "patient-email"
  )[0];

  expect(firstNameInput).toHaveValue("Firstly");
  expect(middleNameInput).toHaveValue("Middleton");
  expect(lastNameInput).toHaveValue("Lastly");
  expect(patientIdInput).toHaveValue("My Custom Id");
  expect(streetOneInput).toHaveValue("My Street 1");
  expect(streetTwoInput).toHaveValue("My Street 2");
  expect(cityInput).toHaveValue("Baltimore");
  expect(countyInput).toHaveValue("Fulton");
  expect(zipCodeInput).toHaveValue("30319");
  expect(phoneInput).toHaveValue("555-555-5555");
  expect(emailInput).toHaveValue("fakeemail@example.com");
};

const labSectionMocks = [
  {
    request: {
      query: GetAllLabsDocument,
    },
    result: {
      data: {
        labs: [
          {
            code: "101289-7",
            description: null,
            display:
              "SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection",
            longCommonName:
              "SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection",
            scaleCode: "LP7751-3",
            scaleDisplay: "Ord",
            systemCode: "LP7633-3",
            systemDisplay: "Thrt",
            answerList: "LL360-9",
            orderOrObservation: "Both",
            panel: false,
          },
          {
            code: "8023-4",
            description: null,
            display:
              "Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay",
            longCommonName:
              "Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay",
            scaleCode: "LP7751-3",
            scaleDisplay: "Ord",
            systemCode: "LP7567-3",
            systemDisplay: "Ser",
            answerList: "",
            orderOrObservation: "Both",
            panel: false,
          },
        ],
      },
    },
  },
  {
    request: {
      query: GetSpecimensByLoincDocument,
      variables: {
        loinc: "LP7567-3",
      },
    },
    result: {
      data: {
        specimens: [
          {
            loincSystemCode: "LP7567-3",
            snomedCode: "122591000",
            snomedDisplay: "Serum specimen from blood product",
            bodySiteList: [],
          },
        ],
      },
    },
  },
];
