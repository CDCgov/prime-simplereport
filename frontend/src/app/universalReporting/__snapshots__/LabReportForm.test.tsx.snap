// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabReportForm LabResultsFormSection renders test result fields when test order and specimen are selected 1`] = `
<div>
  <div
    class="prime-home flex-1"
  >
    <div
      class="grid-container padding-bottom-10 padding-top-4"
    >
      <div
        aria-label="progress"
        class="usa-step-indicator"
        data-testid="step-indicator"
      >
        <ol
          class="usa-step-indicator__segments"
        >
          <li
            class="usa-step-indicator__segment usa-step-indicator__segment--complete"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Facility information
               
              <span
                class="usa-sr-only"
              >
                completed
              </span>
            </span>
          </li>
          <li
            class="usa-step-indicator__segment usa-step-indicator__segment--complete"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Provider information
               
              <span
                class="usa-sr-only"
              >
                completed
              </span>
            </span>
          </li>
          <li
            class="usa-step-indicator__segment usa-step-indicator__segment--complete"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Patient information
               
              <span
                class="usa-sr-only"
              >
                completed
              </span>
            </span>
          </li>
          <li
            aria-current="true"
            class="usa-step-indicator__segment usa-step-indicator__segment--current"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Lab results
               
            </span>
          </li>
          <li
            class="usa-step-indicator__segment"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Review and submit
               
              <span
                class="usa-sr-only"
              >
                not completed
              </span>
            </span>
          </li>
        </ol>
        <div
          class="usa-step-indicator__header"
        >
          <h4
            class="usa-step-indicator__heading"
          >
            <span
              class="usa-step-indicator__heading-counter"
            >
              <span
                class="usa-sr-only"
                data-testid="step-text"
              >
                Step
              </span>
              <span
                class="usa-step-indicator__current-step"
              >
                4
              </span>
               
              <span
                class="usa-step-indicator__total-steps"
              >
                of 5
              </span>
               
            </span>
            <span
              class="usa-step-indicator__heading-text"
            >
              Lab results
            </span>
          </h4>
        </div>
      </div>
      <button
        class="usa-button usa-button--unstyled margin-right-2 margin-bottom-3"
        data-testid="button"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left margin-left-1"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
        Back to 
        patient information
      </button>
      <div
        class="prime-container card-container"
      >
        <div
          class="usa-card__body"
        >
          <div
            class="grid-row"
          >
            <div
              class="grid-col-auto"
            >
              <h3
                class="margin-bottom-0 margin-top-1"
              >
                Test order information
              </h3>
              <p>
                Choose a test order to select specimen type
              </p>
            </div>
          </div>
          <div
            class="grid-row grid-gap"
          >
            <div
              class="grid-col-6 grid-col-mobile"
            >
              <label
                class="usa-legend margin-top-0"
                for="selected-lab"
              >
                Test order
              </label>
              <div
                class="usa-combo-box lab-combo-box usa-combo-box--pristine"
                data-enhanced="true"
                data-testid="combo-box"
              >
                <select
                  aria-hidden="true"
                  class="usa-select usa-sr-only usa-combo-box__select"
                  data-testid="combo-box-select"
                  name="selected-lab"
                  tabindex="-1"
                >
                  <option
                    value="8023-4"
                  >
                    Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay
                  </option>
                  <option
                    value="101289-7"
                  >
                    SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection
                  </option>
                </select>
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-controls="selected-lab--list"
                  aria-describedby="selected-lab--assistiveHint"
                  aria-expanded="false"
                  aria-owns="selected-lab--list"
                  autocapitalize="off"
                  autocomplete="off"
                  class="usa-combo-box__input"
                  data-testid="combo-box-input"
                  id="selected-lab"
                  role="combobox"
                  type="text"
                  value="Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay"
                />
                <span
                  class="usa-combo-box__clear-input__wrapper"
                  tabindex="-1"
                >
                  <button
                    aria-label="Clear the select contents"
                    class="usa-combo-box__clear-input"
                    data-testid="combo-box-clear-button"
                    type="button"
                  >
                     
                  </button>
                </span>
                <span
                  class="usa-combo-box__input-button-separator"
                >
                   
                </span>
                <span
                  class="usa-combo-box__toggle-list__wrapper"
                  tabindex="-1"
                >
                  <button
                    aria-label="Toggle the dropdown list"
                    class="usa-combo-box__toggle-list"
                    data-testid="combo-box-toggle"
                    tabindex="-1"
                    type="button"
                  >
                     
                  </button>
                </span>
                <ul
                  class="usa-combo-box__list"
                  data-testid="combo-box-option-list"
                  hidden=""
                  id="selected-lab--list"
                  role="listbox"
                  tabindex="-1"
                >
                  <li
                    aria-posinset="1"
                    aria-selected="false"
                    aria-setsize="2"
                    class="usa-combo-box__list-option"
                    data-testid="combo-box-option-8023-4"
                    data-value="8023-4"
                    id="selected-lab--list--option-0"
                    role="option"
                    tabindex="-1"
                    value="8023-4"
                  >
                    Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay
                  </li>
                  <li
                    aria-posinset="2"
                    aria-selected="false"
                    aria-setsize="2"
                    class="usa-combo-box__list-option"
                    data-testid="combo-box-option-101289-7"
                    data-value="101289-7"
                    id="selected-lab--list--option-1"
                    role="option"
                    tabindex="-1"
                    value="101289-7"
                  >
                    SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection
                  </li>
                </ul>
                <div
                  class="usa-combo-box__status usa-sr-only"
                  role="status"
                />
                <span
                  class="usa-sr-only"
                  data-testid="combo-box-assistive-hint"
                  id="selected-lab--assistiveHint"
                >
                  When autocomplete results are available use up and down arrows to review
           and enter to select. Touch device users, explore by touch or with swipe
           gestures.
                </span>
              </div>
            </div>
          </div>
          <div
            class="grid-row margin-top-2"
          >
            <div
              class="grid-col-auto"
            >
              <h3
                class="margin-bottom-0 margin-top-5"
              >
                Specimen information
              </h3>
              <p
                class="margin-bottom-0"
              >
                Choose a specimen type to enter test results
              </p>
            </div>
          </div>
          <div
            class="grid-row grid-gap"
          >
            <div
              class="grid-col-6 grid-col-mobile"
            >
              <div
                class="usa-form-group prime-dropdown  card-dropdown"
              >
                <label
                  class="usa-label"
                  for="43"
                >
                  Specimen type
                </label>
                <select
                  aria-required="false"
                  class="usa-select"
                  id="43"
                  name="specimen-type"
                >
                  <option
                    value=""
                  />
                  <option
                    aria-selected="true"
                    value="122591000"
                  >
                    Serum specimen from blood product - 122591000
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="grid-row grid-gap"
          >
            <div
              class="grid-col-6 grid-col-mobile"
            >
              <div
                class="usa-form-group"
              >
                <label
                  class="usa-label"
                  for="44"
                >
                  Collection date
                </label>
                <input
                  aria-required="false"
                  class="usa-input"
                  id="44"
                  max="2025-10-23"
                  min="2020-01-01"
                  name="specimen-collection-date"
                  type="date"
                  value="Invalid date"
                />
              </div>
            </div>
            <div
              class="grid-col-6 grid-col-mobile"
            >
              <div
                class="usa-form-group"
              >
                <label
                  class="usa-label"
                  for="45"
                >
                  Collection time
                </label>
                <input
                  aria-required="false"
                  class="usa-input"
                  id="45"
                  name="specimen-collection-time"
                  step="60"
                  type="time"
                  value="Invalid date"
                />
              </div>
            </div>
          </div>
          <div
            class="grid-row grid-gap"
          >
            <div
              class="grid-col-6 grid-col-mobile"
            >
              <div
                class="usa-form-group prime-dropdown  card-dropdown"
              >
                <label
                  class="usa-label"
                  for="46"
                >
                  Specimen site (optional)
                </label>
                <select
                  aria-required="false"
                  class="usa-select"
                  id="46"
                  name="specimen-collection-body-site"
                >
                  <option
                    aria-selected="true"
                    value=""
                  >
                    No collection location available
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div>
            <div
              class="grid-col-auto"
            >
              <h3
                class="margin-bottom-0 margin-top-5"
              >
                Test result information
              </h3>
              <p>
                The lab order and specimen you selected has one corresponding test result associated with it. Please fill out the test result details below.
              </p>
            </div>
          </div>
          <div
            class="margin-top-2"
          >
            <div
              class="grid-row grid-gap flex-justify"
            >
              <div
                class="grid-col-auto"
              >
                <h4
                  class="margin-bottom-0"
                >
                  Result 
                  1
                   -
                   
                  Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay
                </h4>
              </div>
            </div>
            <div
              class="grid-row grid-gap"
            >
              <div
                class="grid-col-3 grid-col-mobile"
              >
                <div
                  class="usa-form-group"
                >
                  <fieldset
                    class="usa-fieldset prime-radios"
                    data-testid="test-detail-8023-4-test-result-type"
                    id="test-detail-8023-4-test-result-type"
                  >
                    <legend
                      class="usa-legend"
                    >
                      Test result type
                    </legend>
                    <div
                      class=""
                    >
                      <div
                        class="usa-radio"
                      >
                        <input
                          checked=""
                          class="usa-radio__input"
                          data-required="false"
                          id="47-val-ORDINAL"
                          name="test-detail-8023-4-test-result-type"
                          type="radio"
                          value="ORDINAL"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-type-ORDINAL"
                          for="47-val-ORDINAL"
                        >
                          ORDINAL
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="47-val-QUANTITATIVE"
                          name="test-detail-8023-4-test-result-type"
                          type="radio"
                          value="QUANTITATIVE"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-type-QUANTITATIVE"
                          for="47-val-QUANTITATIVE"
                        >
                          QUANTITATIVE
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="47-val-NOMINAL"
                          name="test-detail-8023-4-test-result-type"
                          type="radio"
                          value="NOMINAL"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-type-NOMINAL"
                          for="47-val-NOMINAL"
                        >
                          NOMINAL
                        </label>
                      </div>
                    </div>
                  </fieldset>
                </div>
              </div>
              <div
                class="grid-col-3 grid-col-mobile"
              >
                <div
                  class="usa-form-group"
                >
                  <fieldset
                    class="usa-fieldset prime-radios"
                    data-testid="test-detail-8023-4-test-result-value"
                    id="test-detail-8023-4-test-result-value"
                  >
                    <legend
                      class="usa-legend"
                    >
                      Test result value
                    </legend>
                    <div
                      class=""
                    >
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="48-val-260373001"
                          name="test-detail-8023-4-test-result-value"
                          type="radio"
                          value="260373001"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-value-260373001"
                          for="48-val-260373001"
                        >
                          Positive (+)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="48-val-260415000"
                          name="test-detail-8023-4-test-result-value"
                          type="radio"
                          value="260415000"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-value-260415000"
                          for="48-val-260415000"
                        >
                          Negative (-)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="48-val-455371000124106"
                          name="test-detail-8023-4-test-result-value"
                          type="radio"
                          value="455371000124106"
                        />
                        <label
                          class="usa-radio__label"
                          data-cy="radio-group-option-test-detail-8023-4-test-result-value-455371000124106"
                          for="48-val-455371000124106"
                        >
                          Undetermined
                        </label>
                      </div>
                    </div>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="grid-row grid-gap"
            >
              <div
                class="grid-col-6 grid-col-mobile"
              >
                <div
                  class="usa-form-group"
                >
                  <label
                    class="usa-label"
                    for="49"
                  >
                    Result date
                  </label>
                  <input
                    aria-required="false"
                    class="usa-input"
                    id="49"
                    max="2025-10-23"
                    min="2020-01-01"
                    name="test-detail-8023-4-test-result-date"
                    type="date"
                    value="Invalid date"
                  />
                </div>
              </div>
              <div
                class="grid-col-6 grid-col-mobile"
              >
                <div
                  class="usa-form-group"
                >
                  <label
                    class="usa-label"
                    for="50"
                  >
                    Result time
                  </label>
                  <input
                    aria-required="false"
                    class="usa-input"
                    id="50"
                    name="test-detail-8023-4-test-result-time"
                    step="60"
                    type="time"
                    value="Invalid date"
                  />
                </div>
              </div>
            </div>
            <div
              class="grid-row grid-gap"
            >
              <div
                class="grid-col-6 grid-col-mobile"
              >
                <div
                  class="usa-form-group"
                >
                  <label
                    class="usa-label"
                    data-testid="label"
                    for="test-detail-8023-4-test-result-interpretation"
                  >
                    Notes (optional)
                  </label>
                  <textarea
                    class="usa-textarea interpretation-textarea"
                    data-testid="textarea"
                    id="test-detail-8023-4-test-result-interpretation"
                    maxlength="10000"
                    name="test-detail-8023-4-test-result-interpretation"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="usa-form-group report-form-controls"
          >
            <button
              class="usa-button margin-right-2 margin-top-3"
              data-testid="button"
              type="button"
            >
              Next: 
              Review and submit
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-right "
                data-icon="arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
